guppylang/std/_internal/debug.py

```diff
@@ -43,18 +54,38 @@ def synthesize(self, args: list[ast.expr]) -> tuple[ast.expr, Type]:
         qubit_defn = self.ctx.globals[qubit.id]
         assert isinstance(qubit_defn, TypeDef)
         qubit_ty = qubit_defn.check_instantiate([], self.ctx.globals)
-        for arg in args[1:]:
-            qbt, _ = ExprChecker(self.ctx).check(arg, qubit_ty)
-            if not ExprChecker(self.ctx).check(arg, qubit_ty):
-                raise GuppyTypeError(ExpectedError(arg, "a qubit"))
-            syn_args.append(qbt)
-
-        func_ty = FunctionType(
-            [FuncInput(string_type(), InputFlags.NoFlags)]
-            + [FuncInput(qubit_ty, InputFlags.Inout)] * len(args[1:]),
-            NoneType(),
-        )
+
+        array_len = None
+        arg, ty = ExprSynthesizer(self.ctx).synthesize(args[1])
+        if is_array_type(ty):
+            if len(args) > 2:
+                raise GuppyTypeError(self.MoreThanOneArrayError(self.node))
+            element_ty = get_element_type(ty)
+            if not element_ty == qubit_ty:
+                raise GuppyTypeError(ExpectedError(arg, "an array of qubits"))
+            syn_args.append(arg)
+            func_ty = FunctionType(
+                [
+                    FuncInput(string_type(), InputFlags.NoFlags),
+                    FuncInput(ty, InputFlags.Inout),
+                ],
+                NoneType(),
+            )
+            array_len = get_array_length(ty)
+        else:
+            for arg in args[1:]:
+                qbt, _ = ExprChecker(self.ctx).check(arg, qubit_ty)
+                if not ExprChecker(self.ctx).check(arg, qubit_ty):
+                    raise GuppyTypeError(ExpectedError(arg, "a qubit"))
```

Comment:
```suggestion
```
You're already checking this in the line above. Also `ExprChecker.check` never returns a falsy value